lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1704,22 +1723,43 @@ instance FromJSON ApiAddressData where
         parseBaseAddr v <|>
         parseEnterprise v <|>
         parseRewardAccount v <|>
-        fail "ApiAddressData must have at least one credential."
+        fail msgError
       where
-         parseBaseAddr = withObject "AddrBase" $ \o ->
-             AddrBase <$> o .: "payment" <*> o .: "stake"
-         parseEnterprise = withObject "AddrEnterprise" $ \o ->
-             AddrEnterprise <$> o .: "payment"
-         parseRewardAccount = withObject "AddrRewardAccount" $ \o ->
-             AddrRewardAccount <$> o .: "stake"
+         msgError =
+             "ApiAddressData must have at least one valid credential. When script is\
```

Comment:
The message should not mention `ApiAddressData` actually,  because this only exists for "us" implementors. Clients have no idea about Haskell or about how we internally represent this object.